__tests__/app.test.js

```diff
@@ -72,4 +72,62 @@ describe('/api/articles', () => {
             });
         })
     });
+    test('GET: 400 responds with correct error when given an invalid id', () => {
+        return request(app)
+        .get('/api/articles/invalidid')
+        .expect(400)
+        .then((response) => {
+            const error = response.body;
+            expect(error).toEqual({err: 400, msg: 'Bad request'});
+        })
+    });
+    test('GET: 404 responds with correct error when given an article id that does not exist', () => {
+        const id = 9999
+        return request(app)
+        .get(`/api/articles/${id}`)
+        .expect(404)
+        .then((response) => {
+            expect(response.body).toEqual({ status: 404, msg: 'Page not found.' });
+        })
+    });
+    test('GET: 200 responds with an array of article objects', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            const articles = response.body;
+            expect(articles).toHaveLength(13);
+            articles.forEach((article) => {
+                return expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number),
+                });
+            })
+        })
+    });
+    test('should be sorted by date in descending order', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            const articles = response.body;
+            const dates = articles.map(article => article.created_at);
+            const sortedDates = dates.slice().sort((a, b) => new Date(b) - new Date(a));
```

Comment:
There is library called jest-sorted that will make this much easier but well done for finding your own work around.